app/src/main/java/app/vdh/org/vdhapp/feature/report/data/map/remote/BicyclePathApiClientImpl.kt

```diff
@@ -0,0 +1,84 @@
+package app.vdh.org.vdhapp.feature.report.data.map.remote
+
+import app.vdh.org.vdhapp.core.helpers.CallResult
+import app.vdh.org.vdhapp.feature.report.data.common.remote.ApiRetrofitClient
+import app.vdh.org.vdhapp.core.helpers.safeCall
+import app.vdh.org.vdhapp.feature.report.domain.map.model.BikePathNetwork
+import app.vdh.org.vdhapp.feature.report.domain.map.model.BoundingBoxQueryParameter
+import com.google.android.gms.maps.model.LatLng
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+class BicyclePathApiClientImpl(private val apiClient: ApiRetrofitClient) : BicyclePathApiClient {
+
+    override suspend fun getBicyclePaths(boundingBoxQueryParameter: BoundingBoxQueryParameter?, centerCoordinates: LatLng?, geoJsonItems: JSONArray?, nextToken: String?, network: BikePathNetwork): CallResult<JSONObject> {
+        return safeCall(call = {
+            val latLngQueryParameter = if (centerCoordinates != null) {
+                LatLngQueryParameter(centerCoordinates)
+            } else {
+                null
+            }
+            val result = apiClient.getBicyclePathsAsync(
+                    boundingBoxQueryParameter = boundingBoxQueryParameter,
+                    centerLatLng = latLngQueryParameter,
+                    nextToken = nextToken,
+                    bikePathNetwork = network)
+                    .await()
+
+            if (result.isSuccessful) {
+                val geoJsonResponse = result.body()
+                if (geoJsonResponse != null) {
+                    val jsonObject = JSONObject(geoJsonResponse.string())
```

Comment:
You could use Gson instead of doing this manually ?

Reply:
because the gmaps lib in charge of displaying geojson as polyline takes JsonObject as input ! I know ...